### PR TITLE
Profile bookmarks - Sync Sort Option

### PR DESCRIPTION
**Pull request by ashiagr**
## Description
Adds profile bookmarks sort option sync support between devices.

## Testing Instructions

- Install develop build variant
- Start the app on two different devices (D1, D2) with the same account that needs to have Plus active
- D1: Go to Profile -> Bookmarks
- D1: Set a sort option
- D1: Go back to the profile screen and tap Refresh now
- D2: Go to the profile screen and tap Refresh now
- D2: Open Profile -> Bookmarks
- D2: Ensure that the sort option active is the same as in the D1 device.

## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
 
#### I have tested any UI changes...

- [ ] with different themes
- [ ] with a landscape orientation
- [ ] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/model/BookmarksSortTypeForProfile.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ enum class BookmarksSortTypeForProfile(
     PODCAST_AND_EPISODE(
         labelId = R.string.podcast_and_episode,
         key = SORT_TYPE_PODCAST_AND_EPISODE,
-        serverId = 2,
+        serverId = 4, // Matches the serverId for the same sort type in iOS app
```
Comment (MiSikora) [inline modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/model/BookmarksSortTypeForProfile.kt:29]: AFAIK Web uses `2` as a value.

Comment (ashiagr) [reply modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/model/BookmarksSortTypeForProfile.kt:29]: I checked iOS implementation [here](https://github.com/Automattic/pocket-casts-ios/blob/6a41c5a94dde471f4193593081a5a656ccc6254c/Modules/DataModel/Sources/PocketCastsDataModel/Public/Enums.swift#L79)  which sets podcast and episode sort type value as 4. I'll cross-check again tomorrow and update.

**Review by mebarbosa — APPROVED** [review]
I've tested and it's working as expected

I will leave this as approved since we are waiting to confirm the `serverId` value. Let me know in case you want me to test it again

**Comment by ashiagr** [comment]
Updated `serverId` for the sort order to 2 (p1712900885888749-slack-C06EHRAPW12)

**Pull request merged**